org.eclipse.xtext.common.types/src/org/eclipse/xtext/common/types/util/JvmTypeReferenceUtil.java

```diff
@@ -0,0 +1,56 @@
+/*******************************************************************************
+ * Copyright (c) 2024 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.xtext.common.types.util;
+
+import org.eclipse.emf.common.notify.impl.AdapterImpl;
+import org.eclipse.emf.ecore.util.EcoreUtil;
+import org.eclipse.xtext.common.types.JvmTypeReference;
+
+/**
+ * @author Lorenzo Bettini - Initial contribution and API
+ */
+public class JvmTypeReferenceUtil {
```

Comment:
I don't really like `JvmTypeReferenceUtil` next to `TypeReferences`. Would you think these two classes could be merged (only keeping TypeReferences)?

Reply:
I can put the methods in `TypeReferences`, as instance methods in that case.
I haven't found a `TypeReferencesTest` so I'll create that and move also the tests of `JvmTypeReferenceUtilTest` there

Reply:
> I don't really like `JvmTypeReferenceUtil` next to `TypeReferences`. Would you think these two classes could be merged (only keeping TypeReferences)?

@szarnekow I checked `TypeReferences`, which is based on injection. The new util class doesn't need injection, that's why I created static methods. Does adding the static methods as instance methods in `TypeReferences` still make sense? Or as instance methods? For testing, I seem to understand the common.types.tests don't use injection. I could still create an instance of `TypeReferences` with `new` to test the new methods since I wouldn't use other methods from that class. Just to know whether it still looks fine.

Reply:
You're right. I missed that TypeReferences has quite a few dependencies. The approach based on its own utility class JvmTypeReferenceUtil might be the better option indeed.